roles/openshift_router/tasks/main.yml

```diff
@@ -1,11 +1,14 @@
 ---
-- set_fact: _ortr_images="--images={{ oreg_url|quote }}"
-  when: oreg_url is defined
+
+- set_fact: _ortr_images="--images='{{ openshift.master.registry_url }}'"
+
+- set_fact: _ortr_selector="--selector='{{ openshift.master.router_selector }}'"
 
 - name: Deploy OpenShift Router
   command: >
     {{ openshift.common.admin_binary }} router
-    --create
+    --create --replicas={{ num_infra }}
+    --service-account=router {{ _ortr_selector|default() }}
     --credentials={{ openshift_master_config_dir }}/openshift-router.kubeconfig {{ _ortr_images|default() }}
```

Comment:
I think we should drop the |default() on both the _ortr_selector and _ortr_images in lines 11 & 12 here because in no situation would they be null. They'll always be at least "-images=" and "--selector=".